portal/app/api/v1/endpoint/lora_device.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework.authentication import BasicAuthentication
+from app.api.v1 import CsrfExemptAuth
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from app.models import FarmDevice
+import json
+
+
+class LoraDeviceViewSet(APIView):
+    authentication_classes = (CsrfExemptAuth.CsrfExemptSessionAuthentication, BasicAuthentication)
+
+    def dataparsing(self, data):
```

Comment:
name this `_data_parsing`, the underscore is a soft indicator that this is an internal function.